client/app/bundles/course/assessment/pages/AssessmentIndex/Actions.tsx

```diff
@@ -0,0 +1,122 @@
+import { Create, Inventory, Lock, QuestionMark } from '@mui/icons-material';
+import { Button, IconButton, Tooltip, Typography } from '@mui/material';
+import { AssessmentListData } from 'types/course/assessment/assessments';
+
+import useTranslation, { Descriptor } from 'lib/hooks/useTranslation';
+
+import translations from './translations';
+
+const ACTION_LABELS: Record<AssessmentListData['status'], Descriptor> = {
+  attempting: translations.resume,
+  locked: translations.unlock,
+  open: translations.attempt,
+  submitted: translations.view,
+  unavailable: translations.attempt,
+};
+
+interface ActionsProps {
+  for: AssessmentListData;
+  student: boolean;
+}
+
+const Actions = (props: ActionsProps): JSX.Element => {
+  const { t } = useTranslation();
+
+  const renderUnavailableMessage = (
+    assessment: AssessmentListData,
+  ): JSX.Element => {
+    if (!assessment.isStartTimeBegin)
+      return (
+        <Typography className="italic text-neutral-400" variant="body2">
+          {t(translations.openingSoon)}
```

Comment:
Maybe we can differentiate like if it's less than 1 day then it's opening soon. Otherwise it's not really "soon" if theres an assessment thats 1 month away until the start date.

Reply:
Good point! Unfortunately, I have tried to think of a short phrase (like _Opening soon_) that indicates a later opening date, e.g., _Opening later_, to no avail. I guess I could just harmonise this with the shakey padlock thing and have the opening reason as a tooltip.

This really is a space constraint ><

Reply:
I also made the starting time bold (if it is the unavailable reason) and pulsed on hover.